pkg/proxy/ipvs/proxier_test.go

```diff
@@ -1115,6 +1115,82 @@ func TestClusterIP(t *testing.T) {
 	}
 }
 
+func TestClusterIPIptablesRule(t *testing.T) {
+	svc1 := makeTestService("ns1", "svc1", func(svc *v1.Service) {
+		svc.Spec.ClusterIP = "10.20.30.41"
+		svc.Spec.Ports = []v1.ServicePort{{
+			Name:     "p80",
+			Port:     int32(80),
+			Protocol: v1.ProtocolTCP,
+		}}
+	})
+	ep1 := makeTestEndpoints("ns1", "svc1", func(ept *v1.Endpoints) {
+		ept.Subsets = []v1.EndpointSubset{{
+			Addresses: []v1.EndpointAddress{{
+				IP: "10.180.0.1",
+			}},
+			Ports: []v1.EndpointPort{{
+				Name:     "p80",
+				Port:     int32(80),
+				Protocol: v1.ProtocolTCP,
+			}},
+		}}
+	})
+	ipt := iptablestest.NewFake().SetHasRandomFully(false)
+	ipvs := ipvstest.NewFake()
+	ipset := ipsettest.NewFake(testIPSetVersion)
+	fp := NewFakeProxier(ipt, ipvs, ipset, nil, nil, false)
+	makeServiceMap(fp, svc1)
+	makeEndpointsMap(fp, ep1)
+	fp.syncProxyRules()
+
+	serviceRules := ipt.GetRules(string(kubeServicesChain))
```

Comment:
It might be easier to just test against the entire iptables buffer, as opposed to individual rules/chains.